cross-compilation llvm_12 for aarch64 broken (ncurses linking)

**Describe the bug**

It appears that cross-compiling llvm_12 (with `pkgsCross.aarch64-multiplatform.llvm_12` doesn't work due to an issue where linking ncurses fails.

This is both on staging and staging-next, but not on master, because llvmPackages_latest is bumped to 12.

- [x] Broken on staging (bcc720d9c5ddc6e2ac90370d0be694e3e60c1495)
- [x] Broken on staging-next (d8950b1fad5e542500bb15e9a94d7dda4b16a766) -> probably broken
- [x] Broken on master (997977a64e6501e68543ac9491987556aa38b5c8)

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build -A pkgsCross.aarch64-multiplatform.llvm_12`

**Expected behavior**
A clear and concise description of what you expected to happen.

**Build log**

<details><summary>Error in the build log</summary>

```
[100%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Signals.cpp.o
[100%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/TargetRegistry.cpp.o
[100%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/ThreadLocal.cpp.o
[100%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Threading.cpp.o
[100%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Valgrind.cpp.o
[100%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Watchdog.cpp.o
[100%] Linking CXX static library ../libLLVMSupport.a
[100%] Built target LLVMSupport
Scanning dependencies of target llvm-config
[100%] Building CXX object tools/llvm-config/CMakeFiles/llvm-config.dir/llvm-config.cpp.o
[100%] Linking CXX executable ../../bin/llvm-config
/nix/store/6wfnavjv56bagkd4q81lq5sayy4xf8p8-binutils-2.35.1/bin/ld: /nix/store/7b3w6rdk782yhj3gd4n6mrmwsqzwnhf7-ncurses-6.2-aarch64-unknown-linux-gnu/lib/libtinfo.so: error adding symbols: file in wrong format
collect2: error: ld returned 1 exit status
make[6]: *** [tools/llvm-config/CMakeFiles/llvm-config.dir/build.make:107: bin/llvm-config] Error 1
make[5]: *** [CMakeFiles/Makefile2:26969: tools/llvm-config/CMakeFiles/llvm-config.dir/all] Error 2
make[4]: *** [CMakeFiles/Makefile2:26976: tools/llvm-config/CMakeFiles/llvm-config.dir/rule] Error 2
make[3]: *** [Makefile:7368: llvm-config] Error 2
make[2]: *** [tools/llvm-config/CMakeFiles/NativeLLVMConfig.dir/build.make:80: NATIVE/bin/llvm-config] Error 2
make[1]: *** [CMakeFiles/Makefile2:27192: tools/llvm-config/CMakeFiles/NativeLLVMConfig.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
```

</details>

**Additional context**

This may or may not be related to this change: https://reviews.llvm.org/D85820.

This bump put llvm_12 as a default, but it's most probably not the culprit: 336eb64ae56 (https://github.com/NixOS/nixpkgs/pull/116646). Might be somewhere in the packaging, or upstream.

**Notify maintainers**

@primeos @Ericson2314 @sternenseemann @thefloweringash 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.10-hardened1, NixOS, 21.11pre295280.fa0326ce523 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixos-21.11pre295280.fa0326ce523, nixpkgs-21.11pre293089.1c2986bbb80"`
 - channels(rick): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
attribute:
- pkgsCross.aarch64-multiplatform.llvm_12
```

--- a/pkgs/development/compilers/llvm/12/llvm/default.nix
+++ b/pkgs/development/compilers/llvm/12/llvm/default.nix
@@ -54,7 +54,8 @@ in stdenv.mkDerivation (rec {
   buildInputs = [ libxml2 libffi ]
     ++ optional enablePFM libpfm; # exegesis
 
-  propagatedBuildInputs = [ ncurses zlib ];
+  propagatedBuildInputs = optionals (stdenv.buildPlatform == stdenv.hostPlatform) [ ncurses ]
+    ++ [ zlib ];
 
   patches = [
     ./gnu-install-dirs.patch

